### PR TITLE
DP-25157 fix offered by logic

### PR DESCRIPTION
**Pull request by beto-aveiga**
offered-by-bug

Fixed:
  - project: Patternlab
    component: relationshipIndicators.js
    description: Rewrite relationship indicator folding and toggle logic.
    issue: DP-25157
    impact: Patch

Before:
https://mayflower.digital.mass.gov/patternlab/?p=molecules-relationship-indicators-offered-by-only

After:
https://mayflower.digital.mass.gov/patternlab/b/DP-25157_Fix-breadcrumb-rendering-when-viewport-changes?p=molecules-relationship-indicators-offered-by-only

----

### Logic:
Based on the design notes and what I see on mass.gov, the expected logic should be:

- Desktop: show the first 3 orgs and hide the rest
- Mobile: show the first org and hide the rest

### Examples:
`1 org`: https://www.mass.gov/opinion/cje-opinion-no-89-4
`2 orgs`: https://www.mass.gov/lists/trial-court-zoom-rooms
`3 orgs`: https://www.mass.gov/guidance/revised-guidelines-regarding-covid-19-internal-operating-procedures
`> 3 orgs`: https://www.mass.gov/lists/attorney-court-forms

`2 orgs`:
MF: https://mayflower.digital.mass.gov/patternlab/b/DP-25157_Fix-breadcrumb-rendering-when-viewport-changes/?p=pages-binder-page-cover
Mass.gov: https://www.mass.gov/lists/trial-court-zoom-rooms

`> 3 orgs`:
MF: https://mayflower.digital.mass.gov/patternlab/b/DP-25157_Fix-breadcrumb-rendering-when-viewport-changes/?p=pages-bio-page
Mass.gov: https://www.mass.gov/lists/attorney-court-forms

### CODE REVIEW

File: packages/patternlab/styleguide/source/assets/js/modules/headerTags.js [hunk 1]
```diff
@@ -121,23 +123,29 @@ export default (function (window, document, $, undefined) {
...
+  function resizeResponse() {
+    let windowWidth = $(window).width();
+    groupIndicators(windowWidth < 910 ? 0 : null);
+    // To update the show # more/less button.
+    $('.ma__relationship-indicators--terms .js-relationship-indicator-button').click();
+  }
 
-    }, 50);
+  // Resize events must have a debounced function.
+  $(window).resize(debounce(resizeResponse));
```
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/assets/js/modules/headerTags.js]:
```suggestion
  $(window).resize(debounce(resizeResponse, 50));
```

**Comment by clairesunstudio** [comment]
react backstop failure is a false positive, please ignore

**Comment by clairesunstudio** [comment]
I can't replicate the backstop failure locally. I have run the tests a couple of times. If you can replicate it, please just go ahead and approve it, if not, please ignore. 

Screen Shot 2022-07-26 at 10 11 23 AM

**Review by arthurbaghdas — APPROVED** [review]

**Comment by arthurbaghdas** [comment]
The test was marked as flaky, so I think we are good to go, thank you

**Pull request merged**